zk/poseidon2/src/lib.rs

```diff
@@ -9,6 +9,8 @@ pub type ZkHash = Fr;
 pub trait Digest {
     fn digest(inputs: &[Fr]) -> ZkHash;
 
+    fn compress(inputs: &[Fr; 2]) -> ZkHash;
```

Comment:
It would be nice to add some documentation on the difference between this two calls